scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala

```diff
@@ -779,6 +779,10 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
   def timestampBy(f: T => Instant): SCollection[T] =
     this.parDo(FunctionsWithWindowedValue.timestampFn(f))
 
+  def timestampByWithSkew(f: T => Instant, skew: Duration): SCollection[T] =
+    this.applyTransform(WithTimestamps.of(Functions.serializableFn(f))
```

Comment:
`WithTimestamps.of` is new to me. It wasn't available when I wrote `timestampBy`.

Do you mind do the following:

- add an optional `allowedTimestampSkew` argument (that defaults to `Duration.ZERO`) to `timestampBy` instead of creating a new method
- remove the `FunctionsWithWindowedValue.timestampFn` method
- update ScalaDocs and tests accordingly.